### PR TITLE
Moved modbus transport from addons to core

### PR DESCRIPTION
**Pull request by kaikreuzer**
addresses https://github.com/openhab/openhab-core/issues/1435

Maintainers, please note: I only moved the code and didn't touch it apart from the namespace change - so please refrain from review comments that concern the code itself.

Signed-off-by: Kai Kreuzer <kai@openhab.org>

### CODE REVIEW

File: bundles/org.openhab.core.io.transport.modbus/pom.xml [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+          <bnd><![CDATA[Bundle-SymbolicName: ${project.artifactId}
+Automatic-Module-Name: ${def;bsn}
+Import-Package: \\
+  org.eclipse.jdt.annotation.*;resolution:=optional,\\
+  org.openhab.*;version=!,\\
+  !net.sf.cglib.proxy,\\
+  gnu.io;version="[3.12,6)",\\
+  *
+-sources: false
+-contract: *]]></bnd>
```
Comment (wborn) [inline bundles/org.openhab.core.io.transport.modbus/pom.xml:59]: Would it be possible to move this to a `bnd.bnd` file? That's where we manage these customizations for the other bundles in openhab-core.

**Review by wborn — COMMENTED** [review]
It looks good to me and indeed the code is unchanged except for the packages/imports etc. :+1:  I have one remark below:

File: bundles/org.openhab.core.io.transport.modbus/pom.xml (hunk 1 above)
Comment (kaikreuzer) [reply bundles/org.openhab.core.io.transport.modbus/pom.xml:59]: If you can help me figuring out, what content should be put in that bnd file...? Exactly what is here in the `<bnd>` element?

Comment (kaikreuzer) [reply bundles/org.openhab.core.io.transport.modbus/pom.xml]: Seems to work - done.

Comment (wborn) [reply bundles/org.openhab.core.io.transport.modbus/pom.xml]: Does it also work with one trailing `\`? That's what is used in all the other [bnd.bnd](https://github.com/openhab/openhab-core/blob/master/bundles/org.openhab.core.model.item/bnd.bnd) files. They also all use only one space for indentation.

Comment (kaikreuzer) [reply bundles/org.openhab.core.io.transport.modbus/pom.xml]: sure, done.

**Review by wborn — APPROVED** [review]

**Pull request merged**